### DISCUSSION TITLE
MQTT? Home automation?

### DISCUSSION THREAD

**Discussion by bboti86** [mrbungle64/ecovacs-deebot.js#60] [Q&A] [answered]
Dear @mrbungle64, 

I have a T8 AIVI and also I am a big fan of Home Assistant (www.home-assistant.io). 
Home Assistant has an integration for Ecovacs robots but the underlying library is deprecated (and it seems like it only supports older models). 

Would it be a big struggle to create an integration for your library? The easiest way probably is to wrap somehow into python code or maybe to communicate with an MQTT broker and connect the vacuum over MQTT protocol. 

What do you think? 
B

**Comment by mrbungle64** [comment, marked as answer]
Hello @bboti86 

Have you already tried [this integration](https://github.com/And3rsL/Deebot-for-hassio)?
As far as I know, it should also work for your device.

**Reply by bboti86** [reply]
I feel like super stupid not checking HACS before asking :D 
It works great! Thank and have a Happy New Year!!!

**Reply by mrbungle64** [reply]
I am glad that I could help you. I wish you a Happy New Year too.
